jax/core.py

```diff
@@ -653,6 +653,21 @@ def error(self, *args):
     raise TypeError(concretization_err_msg(fun, context))
   return error
 
+def check_concrete(arg, context: str):
+  """Raises TypeError is the argument contains tracers.
+
+  `arg` can contain scalars, tuple, or arrays."""
+  msg = ("{}: tracer value encountered where a concrete value is expected.\n"
+         "Use `functools.partial` or transformation parameters such as `static_argnums` for `jit` "
+         "to preserve the concrete value during JAX tracing.\n"
+         "See `https://jax.readthedocs.io/en/latest/faq.html#tracer-value-where-concrete-value-is-expected-error`.\n"
+         "Encountered value: {}")
+  if isinstance(arg, Tracer):
+    raise TypeError(msg.format(context, arg))
+  elif isinstance(arg, Sequence):
```

Comment:
Do we need this polymorphism? Maybe we shouldn't handle sequences here, and leave that to the caller.

Reply:
Good point, I can add this feature to the call site. I suspect we'll use this function in more places, we can see later if it needs more functionality.